src/main/java/de/tum/in/www1/exerciseapp/domain/DropLocation.java

```diff
@@ -157,6 +171,22 @@ public DropLocation removeMappings(DragAndDropMapping mapping) {
         return this;
     }
 
+    /**
+     * check if the DropLocation is solved correctly
+     *
+     * @param submittedMappings the new List of submittedMappings from the Result
+     */
+    public boolean isDropLocationCorrect(Set<DragAndDropMapping> submittedMappings) {
+
+        for (DragAndDropMapping submittedMapping : submittedMappings) {
+            if (this.equals(submittedMapping.getDropLocation())
+                && question.getCorrectDragItemsForDropLocation(this).contains(submittedMapping.getDragItem())) {
+                return true;
```

Comment:
I think this method is incorrect.

If this dropLocation is supposed to stay empty, then `question.getCorrectDragItemsForDropLocation(this)` will return an empty list. In this case, we should return true, if `submittedMappings` doesn't include a mapping where `this.equals(submittedMapping.getDropLocation()` returns true, and false, if it does include such a mapping.

As it is now, the method will always return false, if the dropLocation was supposed to stay empty, because the if-condition will never evaluate to true (`.contains()` for an empty List always returns false).

Look at the code in `domain/scoring/ScoringStrategyDragAndDropAllOrNothing.java` (especially lines 22-26).